Robyn/stars

robyng opened the pull request:
swapped star graphics for white background and transparent star shape. Add textarea and options for what the restaurant is good for with value of 1 for each incase we want to calculate a total for future. Still need to figure how to make stars below selected star to change color. Currently only the one we click on changes color. Also need to work on spacing of input fields and size of starts for. Need to refactor styles for styled components, currently using app.css

robyng committed:
add star images, add 5 stars

robyng committed:
Merge branch 'develop' of https://github.com/Healthy-Plate-Project/Healthy-Eating-App into robyn/stars

robyng committed:
removed old star image files, added png transparent star, add textarea for review, add style to textarea in app.css.

robyng committed:
change file name of /Review/index.tsx to /Review/Review.tsx and same for NotFound. Update /Pages/index.tsx file paths

johndavis92790 commented on client/src/App.css:
```diff
@@ -0,0 +1,23 @@
+textarea.paragraph {
+width: 100%;
+height: 150px;
+}
```
These need to go into the associated element styled-components files. I would have @MatteoThomas help you with this.

johndavis92790 commented on client/src/components/nav/Navbar.tsx:
```diff
@@ -6,6 +6,7 @@ const Navbar = () => (
   <NavbarStyled>
     <Link to="/">Home</Link>
     <Link to="results">Results</Link>
+    <Link to="review">Reviews</Link>
```
I would try and keep named things consistant, so this line should be: `<Link to="review">Review</Link>`

johndavis92790 commented on client/src/App.tsx:
```diff
@@ -5,7 +5,8 @@ import {
   Route,
 } from "react-router";
-import  { Home, Results, NotFound }  from './pages'
+import { getRestaurants, getRestaurant } from './utils/serverCalls';
```
No need for this line anymore, these are getting called somewhere else.

robyng replied:
Where are they being called? When I comment out line 8 there are errors looking for Home, Results....

johndavis92790 replied:
It's probably because your branch is behind the current develop branch. You probably need to do a `git pull origin develop` while on your branch, then fix any merge conflicts that come up. I can walk you through it some other time if you want

robyng replied:
Okay I'm still not sure what to do here. Let me know if needs changes.

robyng replied:
To clarify, i believe we do need line 8 `import  { Home, Results, NotFound }  from './pages'` and the develop branch as it is now is still using that code.

johndavis92790 replied:
This comment was for this line, we do not need this in `App.tsx`:
```
import { getRestaurants, getRestaurant } from './utils/serverCalls';
```
not this line, we need this in `App.tsx`:
```
import  { Home, Results, NotFound }  from './pages'
```

johndavis92790 commented on client/src/pages/Review/Review.tsx:
```diff
@@ -0,0 +1,113 @@
+)
+};
+
+export default Review;
```
So it looks like you need to change your tab size in your vscode to 2, rather than 4, I can show you how to do this next time we meet. I would also recommend you turn on the setting to auto format your files so that everything is formatted correctly before you commit the changes. I will actually put together a list of small things like this to go over in our meeting that will save a lot of headaches in the future.

robyng replied:
Thanks, done in our last meeting.

johndavis92790 commented on client/src/pages/Review/Review.tsx:
```diff
@@ -0,0 +1,113 @@
+        <br />
+        <br />
+
+<button type="submit">Submit</button>
```
You can use the universal button element in Button.tsx here, no need to create a new button in this case. You just need to pass in a `content` & `name` prop. @Darolo13 can show you how to do this.

robyng replied:
@Darolo13 yes I need help with this. I looked at how you used Button in your login page but I keep getting typescript error: Type '{ content: string; name: string; }' is not assignable to type 'IntrinsicAttributes'.
[1]   Property 'content' does not exist on type 'IntrinsicAttributes'

robyng replied:
This is what I tried: ` <Button content="Submit" name="test"  />`

johndavis92790 replied:
Without testing myself, I think it's because you might not be able to pass a string directly as a prop like that, you can try to save the string as a variable first, then pass it through, so it would like this:
```
let submit = 'Submit';
let test = 'test';
... 
<Button content={submit} name={test} />
```

robyng replied:
Thanks, not sure why it's working now but it is! <ButtonStyles type='submit'>Submit</ButtonStyles>

johndavis92790 commented on client/src/pages/Review/Review.tsx:
```diff
@@ -0,0 +1,113 @@
+            <p>Please only select if the restaurant offers more than 3 options of that type.</p>
+            <p>Choose all that apply.</p>
+            <label>
+            <input type="checkbox"  name="vegan" aria-label="vegan" id="vegan" value="1"/>
```
I would use something more specific, like `vegan-checkbox` for the `name` and `id`.

johndavis92790 commented on client/src/pages/Review/Review.tsx:
```diff
@@ -0,0 +1,113 @@
+            <label>
+            <input type="radio"  name="star-rating" aria-label="Five Star" id="5-star" value="5"/>
+            <img src={emptyStar} alt="Five Star Rating" />
+            </label>
```
I will look into a good way to do this star rating the way that you are thinking.

robyng replied:
Thanks for the react state solution! Added new photos, outlined star and gold star

robyng replied:
ended up doing a white star instead of an transparent outlined star.

johndavis92790 reviewed (changes requested):
A good start, let me know if you need help with any of this stuff, thanks

johndavis92790 committed:
- Create VeganStarRating.tsx.
- Refactor Review.tsx to have useState's for each of the special diet type star ratings.
- Will use a star rating for each type of special diet.
- Fix App.css selectors to better work with new components.
- Will need to build new files for each type of special diet and push the corresponding useState's into the props of each one.

johndavis92790 committed:
test

robyng committed:
comments in css file

robyng committed:
pull Johns additions for star state and css and VeganStar

robyng committed:
add new star images, remove unused css, update star images in VeganStarReview

robyng committed:
remove old star image and path

johndavis92790 commented on client/src/pages/Review/VeganStarRating.tsx:
```diff
@@ -15,11 +16,11 @@ const VeganStarRating = (props: any) => {
       if (i <= props.selectedVeganStarRating) {
         array.push(
-          <img className="stars checked-star" src={emptyStar} alt={alt} aria-label={alt} key={id} id={id} onClick={starRatingHandler} />
+          <img className="stars checked-star" src={goldStar} alt={alt} aria-label={alt} key={id} id={id} onClick={starRatingHandler} />
```
We don't need the `checked-star` class on this anymore since it doesn't exist anymore.

robyng replied:
Thanks, removed.

johndavis92790 commented on client/src/pages/Review/VeganStarRating.tsx:
```diff
@@ -1,5 +1,6 @@
+import goldStar from "./../../assets/images/gold-star.png";
+import whiteStar from "./../../assets/images/white-star.png";
 
 const VeganStarRating = (props: any) => {
```
Can you duplicate this file for the other special diets on your next commit? Remember you just duplicate and change the names on everything then test, shouldn't be too hard.

johndavis92790 commented on client/src/pages/Review/Review.tsx:
```diff
@@ -14,7 +14,7 @@ const Review = () => {
-      <form className="stars">
+      <form className="starForm">
         <h3>Rate this restaurant based on the special diet:</h3>
         <VeganStarRating
```
And you'll need to duplicate these lines for each special diet also, the useState's are listed above for reference.

johndavis92790 reviewed (changes requested):
Can you also change the white star to not have a black border, I like that look better, similar to the yellow star. I think it will be against a non-white background so it should work fine. Thanks!

robyng committed:
remove checked-star css and class, change textarea css into styled compoentent, import button styles to Review

robyng committed:
add all diets for star reviews, add an overall rating. 100vh for global styles is not enoug for Review page now - it has white bar at bottom and menu overlaps with form.

robyng commented:
@johndavis92790 please review updates to your requested changes. Not sure how to resolve the css issues at this time.

johndavis92790 commented:
> @johndavis92790 please review updates to your requested changes. Not sure how to resolve the css issues at this time.

Send Matthew a message on this to help you out, but as far as I understand it, you put all the css code for a component in the correlated "Styles" .tsx file. For example, all the css code for the "Title" component located in the `Title.tsx` file, would be placed in the `TitleStyles.tsx` file in the same directory.

robyng committed:
remove getRestaurants from app.tsx per John's request. correct spelling of Gluten

robyng committed:
gluten spelling

robyng commented:
Okay all looks good on my end. I would like to make the styling issue on Review a separate tickets because it deals with reworking the Global Styles that @MatteoThomas started.

johndavis92790 commented:
> Okay all looks good on my end. I would like to make the styling issue on Review a separate tickets because it deals with reworking the Global Styles that @MatteoThomas started.

Ya thats fine as long as what you wrote at least works for now.

johndavis92790 committed:
Name fixes for filenames and directories.

johndavis92790 reviewed (approved):
Just a few name fixes for filenames and directories, good job!

johndavis92790 committed:
Merge remote-tracking branch 'origin/develop' into robyn/stars.
Also fixed a couple small issues.
